usr/share/rear/rescue/GNU/Linux/310_network_devices.sh

```diff
@@ -154,22 +206,36 @@ function bridge_handling () {
         # Device is the bridge device
         bridge=$from_network_interface
     else
-        bridge=$(basename $(readlink "/sys/class/net/$from_network_interface/brport/bridge"))
+        bridge=$(basename $(readlink -e "/sys/class/net/$from_network_interface/brport/bridge"))
     fi
     local found=0
     for already_set_up_bridge in $already_set_up_bridges ; do
         test "$bridge" = "$already_set_up_bridge" && found=1 && break
     done
     if [ $found -eq 0 ]; then
         stp=$(cat "/sys/class/net/$bridge/bridge/stp_state")
-        echo "ip link add name $bridge type bridge stp_state $stp" >>$network_devices_setup_script
+        if iplink_has_bridge; then
+            echo "ip link add name $bridge type bridge stp_state $stp" >>$network_devices_setup_script
+        elif [ -x $( which brctl ) ]; then
+            if [[ " ${REQUIRED_PROGS[@]} " != *\ brctl\ * ]]; then
+                REQUIRED_PROGS=( "${REQUIRED_PROGS[@]}" "brctl" )
+            fi
+            echo "brctl addbr $bridge" >>$network_devices_setup_script
+            echo "brctl stp $bridge $stp" >>$network_devices_setup_script
+        else
```

Comment:
Now it is obvious (even for me ;-) why 'brctl' must
be one of the REQUIRED_PROGS because
'brctl' is called during recovery system startup.

Reply:
exactly